### ISSUE TITLE
Applying a texture changes the size of the mesh

### ISSUE THREAD

**Issue by AlexanderJuestel** [pyvista/pyvista#3627]
### Describe the bug, what's wrong, and what you expected.

When applying a texture (image of a seismic line), the cross section/mesh the image is supposed to be put on has shrunk. 

### Steps to reproduce the bug.

```python
import pyvista as pv

p = pv.Plotter(notebook=False)

tex = pv.read_texture('1.png')

# Adding or removing the texture is the only thing that is changed
p.add_mesh(mesh, texture=tex,cmap=odt_petrel)

p.show_grid(color='black')
p.set_background(color='white')
p.set_scale(1,1,1)
p.show()
```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Sun Nov 20 20:19:23 2022 Mitteleuropäische Zeit

                OS : Windows
            CPU(s) : 16
           Machine : AMD64
      Architecture : 64bit
               RAM : 15.2 GiB
       Environment : Jupyter
        GPU Vendor : ATI Technologies Inc.
      GPU Renderer : AMD Radeon(TM) Graphics
       GPU Version : 4.5.14761 Core Profile Context 21.30.40.09 30.0.13040.9001

  Python 3.8.5 (default, Sep  3 2020, 21:29:08) [MSC v.1916 64 bit (AMD64)]

           pyvista : 0.33.3
               vtk : 9.0.1
             numpy : 1.19.2
           imageio : 2.13.1
           appdirs : 1.4.4
            scooby : 0.5.12
        matplotlib : 3.3.2
             PyQt5 : 5.12.3
           IPython : 7.22.0
          colorcet : 3.0.0
             scipy : 1.6.2
              tqdm : 4.50.2
            meshio : 4.4.6

  Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for
  Intel(R) 64 architecture applications
--------------------------------------------------------------------------------
​
```

### Screenshots

Mesh without texture
![image](https://user-images.githubusercontent.com/45469915/202921459-2d52ee86-bd6b-4b92-ac8a-aed6b21324c3.png)

Mesh with texture
![image](https://user-images.githubusercontent.com/45469915/202921493-cec9d2fe-d0da-4312-a9e9-8e307155018f.png)

**Comment by AlexanderJuestel** [comment]
Here also the used mesh and the image
[Mesh.zip](https://github.com/pyvista/pyvista/files/10050505/Mesh.zip)

**Comment by adeak** [comment]
Here's what the image looks like in my image viewer:

![screenshot of image viewer software, showing a checkerboard "transparency" pattern at the top and bottom of the texture image](https://user-images.githubusercontent.com/17914410/204671908-408a0baa-1f93-4b72-a9a7-813b80c18060.png)

Note the checkerboard strips at the top and bottom. Isn't it just that your image has transparent parts, and those don't get rendered on your mesh? You can check

**Comment by banesullivan** [comment]
Yeah, the opacity in the texture is affecting the opacity of the mesh. I feel like this might have behaved differently in the past but cannot confirm. I see the same issue in ParaView and across a few PyVista versions. I think we should check this against VTK versions to see if texture rendering changed.

```py
import pyvista as pv
import numpy as np

mesh = pv.read('/Users/bane.sullivan/Downloads/Mesh/Mesh.vtk')
tex = pv.read_texture('/Users/bane.sullivan/Downloads/Mesh/1.png')

p = pv.Plotter(notebook=False)

p.add_mesh(mesh, texture=tex)
p.add_mesh(mesh, color='red', opacity=0.5)

p.show_grid(color='black')
p.set_background(color='white')
p.set_scale(1,1,1)
p.show()
```

<img width="624" alt="Screen Shot 2022-12-02 at 9 57 00 AM" src="https://user-images.githubusercontent.com/22067021/205345042-91ae6f71-8618-4ecd-9189-1fbc4d7815b5.png">

**Comment by AlexanderJuestel** [comment]
Yeah, so cropping the image does the trick. Just wondered why that is. The image was exported using numpy....

![image](https://user-images.githubusercontent.com/45469915/205359661-34ffc890-7eb5-4288-beef-82db8977b5dc.png)

**Comment by AlexanderJuestel** [comment]
Yeah, using the following, I always ended up with this green screen for some reason

```ruby
import pyvista as pv

tex = pv.numpy_to_texture(data)

p = pv.Plotter(notebook=False)

p.add_mesh(mesh, texture=tex, cmap='seismic', clim=[-5,5])

p.show_grid(color='black')
p.set_background(color='white')
p.show()
```

![image](https://user-images.githubusercontent.com/45469915/205375221-314f9790-6713-4efd-b445-fa62b287fcdd.png)

**Comment by AlexanderJuestel** [comment]
Got it using your draping example: 

![image](https://user-images.githubusercontent.com/45469915/205377684-efef8435-1616-4fa7-9e75-a7bed88002bf.png)

```ruby


```

https://docs.pyvista.org/examples/00-load/create-surface-draped.html#sphx-glr-examples-00-load-create-surface-draped-py

**Comment by AlexanderJuestel** [comment]
So, after adjusting the z_spacing and the z position, it is also possible to like change the clims of the mesh. 

That is amazing :) 
![image](https://user-images.githubusercontent.com/45469915/205380776-e10accf6-24ca-4b67-98cb-a3cd8ef1064b.png)

![image](https://user-images.githubusercontent.com/45469915/205380698-ef67b425-4d92-4d58-adf9-58383882ff51.png)
